console/console-init/ui/src/modules/iot-device/components/AddDeviceWithJson/AddDeviceWithJson.tsx

```diff
@@ -114,85 +123,145 @@ const AddDeviceWithJson: React.FunctionComponent<IAddDeviceWithJsonProps> = ({
         variant="danger"
         isInline
-        visible={showJsonValidationError}
+        visible={errorMessage !== ""}
         title="Error"
-        description="Invalid JSON syntax, unable to parse JSON"
+        description={errorMessage}
       />
       <br />
       <Grid>
-        <GridItem span={9} className={css(styles.box_align_style)}>
-          <JsonEditor
-            value={deviceDetail}
-            readOnly={false}
-            name={"editor-add-device"}
-            setDetail={setDeviceInfoInDetail}
-            style={{
-              minHeight: "39em"
-            }}
-          />
-        </GridItem>
-        <GridItem span={3} className={css(styles.box_align_style)}>
-          <PageSection variant={PageSectionVariants.light}>
-            <AddJsonUsingTemplate
-              setDetail={setDeviceInfoInDetail}
-              selectedTemplate={selectedTemplate}
-              setSelectedTemplate={setSelectedTemplate}
-            />
-          </PageSection>
-        </GridItem>
+        {isPreviewEnabled ? (
+          <div className={css(styles.padding_left)}>
+            <ReviewDeviceContainer device={getDeviceDetail()} />
+          </div>
+        ) : (
+          <>
+            <GridItem span={9} className={css(styles.box_align_style)}>
+              <JsonEditor
+                value={deviceDetail}
+                readOnly={false}
+                name={"editor-add-device"}
+                setDetail={setDeviceInfoInDetail}
+                style={{
+                  minHeight: "39em"
+                }}
+              />
+            </GridItem>
+            <GridItem span={3} className={css(styles.box_align_style)}>
+              <PageSection variant={PageSectionVariants.light}>
+                <AddJsonUsingTemplate
+                  setDetail={setDeviceInfoInDetail}
+                  selectedTemplate={selectedTemplate}
+                  setSelectedTemplate={setSelectedTemplate}
+                  setErrorMessage={setErrorMessage}
+                />
+              </PageSection>
+            </GridItem>
+          </>
+        )}
       </Grid>
       <br />
-
-      <Button
-        id="add-device-json-finish-button"
-        aria-label="Finish button"
-        variant="primary"
-        onClick={onFinish}
-      >
-        Finish
-      </Button>
-      {"  "}
-      <Button
-        id="add-device-json-preview-button"
-        aria-label="Preview button"
-        variant="secondary"
-        onClick={handleOnPreview}
-      >
-        Preview
-      </Button>
-      {"  "}
-      <Button
-        id="add-device-json-cancel-button"
-        aria-label="cancel button"
-        variant="link"
-        onClick={onCancel}
-      >
-        Cancel
-      </Button>
+      <Split className={css(styles.padding_left)} hasGutter>
+        <SplitItem>
+          <Button
+            id="add-device-json-finish-button"
+            aria-label="Finish button"
```

Comment:
upper case and lower case don't matter for aria-label so it would better to use lower case for all aria tags.